File: src/mainWindow.cpp
#include "mainWindow.hpp"

#include <QPixmap>
#include <QLabel>

#include <string>

MainWindow::MainWindow(
	QWidget* parent, 
	Qt::WindowFlags flags
): QMainWindow(parent, flags) {
	setWindowTitle("Title");

	m_lay->setContentsMargins(0,0,0,0);
	m_central_widget->setLayout(m_lay);
	setCentralWidget(m_central_widget);

	auto inputLay = new QHBoxLayout;
	m_lay->addLayout(inputLay);
	inputLay->setContentsMargins(10,5,10,5);

	connect(
		m_solve_button, &QPushButton::pressed,
		this, &MainWindow::solve
	);
}

void MainWindow::solve() {

}

File: src_py/main_window.py
from PySide6.QtWidgets import (
    QMainWindow, 
    QWidget, QLayout, QHBoxLayout, QVBoxLayout,
    QLabel, QTextEdit, QSpinBox,
    QPushButton
)

class main_window(QMainWindow):
    def __init__(self):
        QMainWindow.__init__(self)
        self.main_widget = QWidget()
        self.main_layout = QVBoxLayout() 
        self.main_widget.setLayout(self.main_layout)
        self.setCentralWidget(self.main_widget)

        self.solve_button = QPushButton()
        self.solve_button.clicked.connect(self.solve)

        self.main_layout.addWidget(QLabel("Hello, world!"))

    def solve(self):
        pass
